src/SNLS_TrDLDenseG.h

```diff
@@ -197,239 +165,62 @@ class SNLSTrDlDenseG
-                     //
-                     double beta = (-qb+sqrt(qb*qb-4.0*qa*qc))/(2.0*qa) ;
-#ifdef SNLS_DEBUG
-                     if ( beta > 1.0 || beta < 0.0 ) {
-                        SNLS_FAIL(__func__, "beta not in [0,1]") ;
-                     }
-#endif
-                     beta = fmax(0.0,fmin(1.0,beta)) ; // to deal with and roundoff
-
-                     // delx[iX] = alpha*ngrad[iX] + beta*p[iX] = beta*nrStep[iX] - (1.0-beta)*alpha*grad[iX]
-                     //
-                     {
-                        double omb  = 1.0-beta ;
-                        double omba = omb*alpha ;
-                        for (int iX = 0; iX < _nDim; ++iX) {
-                           delx[iX] = beta * nrStep[iX] - omba*grad[iX] ;
-                        }
-                        pred_resid = omb * res_cauchy ;
-                     }
-
-#ifdef __cuda_host_only__
-                     if ( _os != nullptr ) {
-                        *_os << "trying step along second leg" << std::endl ;
-                     }
-#endif
-                  } // if norm_s_sd_opt >= delta
-
-               } // use_nr
-               
-               this->update( delx ) ;
-               reject_prev = false ;
-               //
-               {
-                  bool rjSuccess = this->computeRJ(residual, Jacobian) ; // at _x
-                  if ( !(rjSuccess) ) {
-                     // got an error doing the evaluation
-                     // try to cut back step size and go again
-                     bool deltaSuccess = _deltaControl->decrDelta(_os, _delta, nr2norm, use_nr ) ;
-                     if ( ! deltaSuccess ) {
-                        _status = deltaFailure ;
-                        break ; // while ( _nIters < _maxIter ) 
-                     }
-                     reject_prev = true ;
-                  }
-                  else {
-                     _res = this->normvec(residual) ;
-#ifdef __cuda_host_only__
-                     if ( _os != nullptr ) {
-                        *_os << "res = " << _res << std::endl ;
-                     }
-#endif
-
-                     // allow to exit now, may have forced one iteration anyway, in which
-                     // case the delta update can do funny things if the residual was
-                     // already very small 
+               bool rjSuccess = this->computeRJ(residual, Jacobian) ; // at _x
+               // Could also potentially include the reject previous portion in here as well
+               // if we want to keep this similar to the batch version of things
+               snls::updateDelta<_nDim>(_deltaControl, residual, res_0, pred_resid, nr_norm, _tolerance, use_nr, rjSuccess,
+                                        _delta, _res, _rhoLast, reject_prev, _status, _os);
+               // This new check is required due to moving all the delta update stuff into its own function to share features between codes
```

Comment:
We currently don't have any formatting conventions within SNLS. We could just copy the uncrustify file used within in ExaCMech temporarily. I was actually thinking this might be something we might want to add.